From b1ec9f808e745054cdd57ee8bb242eba0cca078a Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Fri, 10 Aug 2018 17:07:47 +0200
Subject: [PATCH 2/4] soong: Enforce absolute path if OUT_DIR is set

 * Follows "soong: use absolute path for OUT_DIR"
    to resolve inline kernel building when
    export OUT_DIR=out or =out_name

Change-Id: I6560ec163cc6508f211bb1cd65d565015141d79b
---
 ui/build/config.go | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ui/build/config.go b/ui/build/config.go
index a54ed82b..f95eb646 100644
--- a/ui/build/config.go
+++ b/ui/build/config.go
@@ -65,7 +65,11 @@ func NewConfig(ctx Context, args ...string) Config {
 
 	// Make sure OUT_DIR is set appropriately
 	if outDir, ok := ret.environ.Get("OUT_DIR"); ok {
-		ret.environ.Set("OUT_DIR", filepath.Clean(outDir))
+		outDir := filepath.Clean(outDir)
+		if (!filepath.IsAbs(outDir)) {
+			outDir = filepath.Join(os.Getenv("TOP"), outDir)
+		}
+		ret.environ.Set("OUT_DIR", outDir)
 	} else {
 		outDir := "out"
 		if baseDir, ok := ret.environ.Get("OUT_DIR_COMMON_BASE"); ok {
-- 
2.17.1

